File: Ponto.rb
class Ponto

attr_reader :x, :y

  def initialize(x,y)
    @x = x
    @y = y
  end

  def consultar_x()
    @x
  end

   def consultar_y()
    @y
  end

  def ==(outro)
    x == outro.x && y == outro.y
  end
end

File: Ponto.java

public class Ponto {

	private double x;
	private double y;

	public Ponto(double x, double y) {
		this.x = x;
		this.y = y;
	}

	public double x() {
		return this.x;
	}
	
	public double y() {
		return this.y;
	}
	
	public boolean equals(Object outro){
		Ponto outroPonto = (Ponto) outro;
		return this.x == outroPonto.x && this.y == outroPonto.y;
	}

}
